instrumentation/opentelemetry/span_test.go

```diff
@@ -109,3 +109,24 @@ func TestGetAttributes(t *testing.T) {
 	assert.Equal(t, "string_value", attrs.GetValue("string_key"))
 	assert.Equal(t, nil, attrs.GetValue("non_existent"))
 }
+
+func TestGetAllAttributes(t *testing.T) {
+	sampler := sdktrace.AlwaysSample()
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sampler),
+	)
+	otel.SetTracerProvider(tp)
+	_, s, _ := StartSpan(context.Background(), "test_span", &sdk.SpanOptions{})
+	s.SetAttribute("k1", "v1")
+	s.SetAttribute("k2", 200)
+	attrs := s.GetAttributes().GetAll()
+
+	assert.Equal(t, 3, len(attrs))
```

Comment:
Why 3?

Reply:
`service.instance.id` is already there as an attribute

Reply:
```suggestion
        // service.instance.id is added implicitly in StartSpan so 3 attributes will be present.
	assert.Equal(t, 3, len(attrs))
```